### ISSUE TITLE
Support for multiple types of geocoder (Local and Remote)

### ISSUE THREAD

**Issue by chriscalip** [perliedman/leaflet-control-geocoder#123]
I understand support for multiple remote geocoder is another task see https://github.com/perliedman/leaflet-control-geocoder/issues/121.

It is possible to have multiple geocoder callbacks
If and only if result callback is available from remote; then call local geocoder for further results.

For autocomplete functionality .. can be available with a local geocoder.

In my case I have a local fuse-search https://github.com/krisk/Fuse doing autocomplete and on submit call doing both remote geocoder and local fuse-search.

Is this worthwhile for this project?

**Comment by perliedman** [comment]
Since all geocoders must be asynchronous, I don't think there's any real distinction between "remote" and "local" geocoders from the control's point of view.

What I think might be the difference in the approach your describing, compared to #121, is that your approach wants to first call geocoder A, and only call geocoder B if A did not return any results, effectively calling the geocoders serially. The approach in #12, on the other hand, calls all geocoders in parallel and aggregates the results from them.

I can see use cases for both approaches, and I don't think the implementation would be very complex. PRs are welcome.

**Comment by chriscalip** [comment]
Perhaps geocoders is not the right word; but a local source of geojson locations.
Well in my use case.. i had a geojson array (name,city,state,postal code, etc..) already in the html document at page load; these bunch of locations were painted to the map as markers.  On submit event it given the search text; after getting the results from google; gets results from fuse-search on search text to the local geojson array.
The results are merged together.

**Comment by tijmenheid** [comment]
Hi first of all thanks for the great work, I'ts working like a charm! I came upon this issue and am looking for a function very close to this one, it might be a great add-on in general since there are a lot of people needing a geolocate and menu-search option on the same page.

Issue: Right now I have two autosuggest search fields. 
1 - for a menu with places/pois ('bars', 'museums', 'supermarkets' etc.) all tucked in an array with key-words and query-url, placed in the index file. 
2 - your leaflet geocoder.

It would be great to have an option to merge those 2 textfields into 1 by having an option to 'add values' or something similar/simple, which basically would be just sticking a second string to the result of the main geo-query (coming back from for example mapbox) and then letting it roll through the rest of the code as usual. What do you think?
